src/main.cpp

```diff
@@ -168,9 +203,77 @@ namespace
         }
+        return CODE(RULE_ENGINE_CONTINUE);
+    }
+
+    auto exec_rule_atomic_metadata(std::list<boost::any>& _rule_arguments, irods::callback _effect_handler) -> irods::error 
+    {
+        auto is_input_valid = [](const bytesBuf_t* _input) -> std::tuple<bool, std::string>
+        {
+            if (!_input) {
+                return {false, "Missing JSON input"};
+            }
+
+            if (_input->len <= 0) {
+                return {false, "Length of buffer must be greater than zero"};
+            }
+
+            if (!_input->buf) {
+                return {false, "Missing input buffer"};
+            }
 
+            return {true, ""};
+        };
+
+        try {
+            auto* input_bb = boost::any_cast<bytesBuf_t*>(*std::next(std::begin(_rule_arguments), 2));
+            auto& rei = get_rei(_effect_handler);
+            const auto config = load_plugin_config(rei);
+
+
+            if (const auto [valid, msg] = is_input_valid(input_bb); !valid) {
+                log_re::error({{"log_message", msg}});
+                return ERROR(INPUT_ARG_NOT_WELL_FORMED_ERR, msg);
+            }
+
+            json input = json::parse(std::string(static_cast<const char*>(input_bb->buf), input_bb->len));
+
+            const auto& operations = input.at("operations");
+            std::vector<irods::error> op_codes; 
+
+            std::transform(operations.begin(), operations.end(), std::back_inserter(op_codes), [&rei](json op) {
+                return rule_logic(op.at("attribute").get_ref<const std::string&>(), rei);
+            });
+            if(const auto& ret = std::find_if(op_codes.begin(), op_codes.end(), [&op_codes](const irods::error& err) { return !err.ok(); }); ret != op_codes.end()) {
+                return *ret;
+            }
+                return CODE(RULE_ENGINE_CONTINUE);
+        }
+        catch (const json::parse_error& e) {
+            log_re::error({{"log_message", "Failed to parse input into JSON"}, {"error_message", e.what()}});
+        }
+        catch (const std::exception& e) {
+            log_re::error({{"log_message", e.what()}, {"rule_engine_plugin", "metadata_guard"}});
+        }
         return CODE(RULE_ENGINE_CONTINUE);
     }
+
+    
+    auto exec_rule(irods::default_re_ctx&,
+                   const std::string& _rule_name,
+                   std::list<boost::any>& _rule_arguments,
+                   irods::callback _effect_handler) -> irods::error
+    {
+        std::unordered_map<std::string, std::function<irods::error(std::list<boost::any>&, irods::callback )>> lookup_table{
+            {"pep_api_mod_avu_metadata_pre",  exec_rule_modavu},
+            {"pep_api_atomic_apply_metadata_operations_pre", exec_rule_atomic_metadata}};
+        if(lookup_table.contains(_rule_name)) {
+            return lookup_table.at(_rule_name)(_rule_arguments, _effect_handler);
+        }
+        else {
+            return CODE(RULE_ENGINE_CONTINUE);
+        }
```

Comment:
No need for `else` here because of the `return` statement in the if-block above.

```cpp
if (cond) {
    return work();
}

return the_default;
```